compiler/rustc_mir_build/src/build/matches/match_pair.rs

```diff
@@ -42,7 +42,12 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
         let tcx = self.tcx;
         let (min_length, exact_size) = if let Some(place_resolved) = place.try_to_place(self) {
             match place_resolved.ty(&self.local_decls, tcx).ty.kind() {
-                ty::Array(_, length) => (length.eval_target_usize(tcx, self.param_env), true),
+                ty::Array(_, length) => (
+                    length
+                        .try_to_target_usize(tcx)
+                        .expect("expected len of array pat to be definite"),
```

Comment:
"definite"?

Reply:
like... a definite, evaluatable numerical value.

Reply:
huh i dont think ive ever seen that word used like that

Reply:
If you have a better suggestion then LMK lol